Scrabble.java

```diff
@@ -85,16 +103,29 @@ public static void playHand(String hand) {
 			// non-whitespace characters. Whitespace is either space characters, or  
 			// end-of-line characters.
 			String input = in.readString();
-			//// Replace the following break statement with code
-			//// that completes the hand playing loop
-			break;
-		}
-		if (hand.length() == 0) {
-	        System.out.println("Ran out of letters. Total score: " + score + " points");
-		} else {
-			System.out.println("End of hand. Total score: " + score + " points");
+			if (input.equals(".")) {
+				break;
+			}
+			if (MyString.subsetOf(input, hand) == true && isWordInDictionary(input) == true) {

```

Comment:
1. split into 2 and handle negatives first
2. consider the following
`if (MyString.subsetOf(input, hand) && isWordInDictionary(input))`
by comparing to true you are taking non required steps of calc which make you code less readable.. 

for false cases. 
`if (!isWordInDictionary(input))`